asdf_astropy/converters/transform/tests/test_transform.py

```diff
@@ -0,0 +1,449 @@
+import asdf
+import pytest
+import numpy as np
+
+from astropy.modeling import models as astropy_models
+import astropy.modeling
+from astropy import units as u
+from numpy.testing import assert_array_equal
+from asdf.tests.helpers import assert_roundtrip_tree, yaml_to_asdf
+
+
+# TODO: Uncomment this fixture once astropy converters
+# have been fully implemented.
+# @pytest.fixture(autouse=True)
+# def remove_astropy_extensions():
+#     """
+#     Disable the old astropy extension so that it doesn't
+#     confuse our test results.
+#     """
+#     with asdf.config_context() as config:
+#         config.remove_extension(package="astropy")
+#         yield
+
+
+def assert_model_roundtrip(model, tmpdir, version=None):
+    """
+    Assert that a model can be written to an ASDF file and read back
+    in without losing any of its essential properties.
+    """
+    path = str(tmpdir / "test.asdf")
+
+    with asdf.AsdfFile({"model": model}, version=version) as af:
+        af.write_to(path)
+
+    with asdf.open(path) as af:
+        assert_models_equal(model, af["model"])
+        return af["model"]
+
+
+def assert_models_equal(a, b):
```

Comment:
Would it make sense to add this as a `__eq__` method to `astropy.modeling` core?

Also, there's a lot of specific comparisons in this assertion helper, which is good.  I like that it is explicit.  But is there a reason to not use `modela.__dict__ == modelb.__dict__` as the comparison?  Does that not capture the underlying structure of the object?  Or is it better to compare properties instead of the underlying variable holding the value?  For example:

```
In [4]: m = models.Rotation2D()

In [5]: m.__dict__
Out[5]: 
{'_inputs': ('x', 'y'),
 '_outputs': ('x', 'y'),
 '_name': None,
 'angle': Parameter('angle', value=0.0),
 '_mconstraints': {'eqcons': [], 'ineqcons': []},
 '_model_set_axis': False,
 '_param_metrics': defaultdict(dict,
             {'angle': {'slice': slice(0, 1, None), 'shape': (), 'size': 1}}),
 '_n_models': 1,
 '_parameters': array([0.]),
 '_input_units_strict': {'x0': False, 'x1': False},
 '_input_units_allow_dimensionless': {'x0': False, 'x1': False}}
```

Reply:
> Would it make sense to add this as a __eq__ method to astropy.modeling core?

I thought someone warned me that this would be controversial, but I don't remember clearly.  We don't need the models to be equal in all respects since there are some attributes that we don't serialize, so it may be better just to check what we know we need.

> But is there a reason to not use modela.__dict__ == modelb.__dict__ as the comparison?

I didn't try that, but comparing `__dict__` requires that objects at all levels have a reasonable `__eq__` implementation.  I don't see an `__eq__` implementation for `Parameter`, for example.

Reply:
Ah, ok.  That's a problem.  So yeah.  I guess this cascades.  If we want to compare things, things need `__eq__` defined.